packages/ra-ui-materialui/src/button/PrevNextButton.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react';
+import {
+    useCreatePath,
+    useGetList,
+    useGetRecordId,
+    useListParams,
+    useResourceContext,
+    useTranslate,
+} from 'ra-core';
+import { NavigateBefore, NavigateNext } from '@mui/icons-material';
+import { Link } from 'react-router-dom';
+import { IconButton } from '@mui/material';
+
+export const PrevNextButton = (props: PrevNextButtonProps) => {
+    const { linkType = 'edit' } = props;
+    const translate = useTranslate();
+    const recordId = useGetRecordId();
+    const resource = useResourceContext();
+    const createPath = useCreatePath();
+    const [query] = useListParams({
+        resource,
+    });
+    const list = useGetList(resource, {
+        pagination: { page: query.page, perPage: query.perPage },
```

Comment:
The pagination shouldn't depend on the params - you always want to fetch * all* records. You may want to have a `perPage` prop fixed at an arbitrary level though (e.g. 1000) to avoid exploding the app when there are too many records.